services/config_service.rb

```diff
@@ -33,14 +33,14 @@ def active_code_hosting_service_key(provider_credential: nil)
     def code_hosting_service(provider_credential: nil)
       code_hosting_service_key = active_code_hosting_service_key(provider_credential: provider_credential)
       code_hosting_service = self.active_code_hosting_services[code_hosting_service_key]
-      return code_hosting_service unless code_hosting_service.nil?
+      return code_hosting_service if code_hosting_service.kind_of?(FastlaneCI::CodeHostingService)
 
       # TODO: not a big deal right now, but we should have a way of automatically generating the correct
       # CodeHostingService subclass based on the provider_credential type and maybe not have it right here.
       # A Java-style factory might be the right move here.
       case provider_credential.type
       when FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
-        code_hosting_service = GitHubService.new(provider_credential: provider_credential)
+        code_hosting_service = FastlaneCI::GitHubService
```

Comment:
this is weird, what's going on here? Do you mean to access the `Class` instead of an instance of the class?

Reply:
So `GitHubService` class methods are intended to be called out of the `project`'s scope. So whenever we want to operate without a `Project` in the scope, it's all being done through its class methods. So this is done in purpose, I want to access to the `Class` itself.

Reply:
My future review is to create a mixin that provides a unified interface for all `GitHubService`, `BitbucketService`, etc.

Reply:
Ah, ok, `code_hosting_service` implies to me that you're storing an object instance reference here, not a class reference. Maybe you can change the name to `code_hosting_service_class` and that will help clear up my confusion 😄